packages/taco/src/taco-config-validator.ts

```
@@ -0,0 +1,362 @@
+  const domainInfo = Object.values(DOMAINS).find(
+    (info) => info.domain === config.domain,
+  );
+  return !!domainInfo && domainInfo.chainId === config.chainId;
+}
+
+/**
+ * Generic validation result interface
+ */
+export interface ValidationResult {
+  isValid: boolean;
+  errors: string[];
+}
+
+/**
+ * TACo Configuration Validator
+ *
+ * Validates TACo client configurations, domains, and provider compatibility.
+ * Provides both fast and full validation methods for TACo operations.
+ */
+export class TacoConfigValidator {
+  /**
+   * Get all supported TACo domain names
+   * @returns {DomainName[]} Array of supported TACo domain names ('lynx', 'tapir', 'mainnet')
+   */
+  static getSupportedDomains(): DomainName[] {
+    return Object.values(DOMAINS).map((domain) => domain.domain);
+  }
+
+  /**
+   * Check if domain is valid
+   * @param {DomainName} domain - TACo domain name to check ('lynx', 'tapir', 'mainnet')
+   * @returns {boolean} True if domain exists
+   */
+  static isValidDomain(domain: DomainName): boolean {
+    return !!domain && this.getSupportedDomains().includes(domain);
+  }
+
+  /**
+   * Get expected chain ID for domain from DOMAINS configuration
+   * @param {DomainName} domain - Domain name to look up
+   * @returns {number | undefined} Chain ID for the domain, undefined if not found
+   * @private
+   */
+  private static getExpectedChainId(domain: DomainName): number | undefined {
+    const domainEntry = Object.values(DOMAINS).find(
+      (domainConfig) => domainConfig.domain === domain,
+    );
+    return domainEntry?.chainId;
+  }
+
+  /**
+   * Validate ritual ID (basic validation - positive number only)
+   * @param {DomainName} domain - Domain name (unused but kept for API compatibility)
+   * @param {number} ritualId - Ritual ID to validate
+   * @returns {boolean} True if valid (positive number)
+   */
+  static isValidRitualId(domain: DomainName, ritualId: number): boolean {
```

Since this is purely checking whether ritual id is a number and non-positive, perhaps the name of the function should reflect that a bit better, perhaps `isValidRitualIdNumber` or something like that.

---

Actually, it should be a positive integer. I updated the code.

However, in all cases, I think the criteria of a valid ritual id (it is a number or so) is-not/should-not be the concern of the code calling this function, except passing the correct type. As for the rest of the code this function checks the validity of the ritual id. And only this function knows and concerned about what the specifications of a valid id are, even it is now just a positive integer.

---

+1 on this